File: exercises/modelling_in_code/typescript/test/playground.test.ts
import { describe, it, expect } from "bun:test"
import { Application } from "../src/Application"

// Feel free to use this test file to try out your objects.
// You can even test-drive them, if you wish!
// Run `bun test --watch` in the terminal to run the tests.

it("works", () => {
  expect(1 + 1).toBe(2);
});

describe("Application", () => {
  it("refuses to construct without id and receivedDateTime", () => {
    expect(() => new Application(null as any, null as any)).toThrow();
  });
});

File: exercises/ddd_infrastructure_layer/java/src/test/java/PlaygroundTest.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

import domain.application.Application;

// Feel free to use this test file to try out your objects.
// You can even test-drive them, if you wish!
// Run `gradle test --continuous` in the terminal to run the tests.

public class PlaygroundTest {
    @Test
    void testWorks() {
        assertEquals(1 + 1, 2 );
    }

    @Test
    void testRefusesToConstructWithoutIdAndReceivedDateTime() {
        assertThrows(RuntimeException.class, () -> {
            new Application(null, null);
        });
    }
}
